python_modules/dagster/dagster/_core/execution/context/compute.py

```diff
@@ -410,6 +410,20 @@ def add_metadata_two_outputs(context) -> Tuple[str, int]:
             metadata=metadata, output_name=output_name, mapping_key=mapping_key
         )
 
+    def merge_output_metadata(
+        self,
+        metadata: Mapping[str, Any],
+        output_name: Optional[str] = None,
+        mapping_key: Optional[str] = None,
+    ) -> None:
+        metadata = check.mapping_param(metadata, "metadata", key_type=str)
+        output_name = check.opt_str_param(output_name, "output_name")
+        mapping_key = check.opt_str_param(mapping_key, "mapping_key")
+
+        self._step_execution_context.add_output_metadata(
+            metadata=metadata, output_name=output_name, mapping_key=mapping_key, merge=True
+        )
+
```

Comment:
can you add an `_add_output_metadata` helper method that takes the bool and leave the public one unmodified?

Reply:
`StepExecutionContext.add_output_metadata` isn't public, it's accessed through the public `OpExecutionContext.add_output_metadata`, which doesn't have any modifications in this PR.

Reply:
Or do you mean public just in the Python (not user-facing) sense? `StepExecutionContext.add_output_metadata` while python-public is an internal API so it seems find to add a switch?